vulkan_generator/struct_factories/generator.py

```diff
@@ -0,0 +1,451 @@
+            typedef uint64_t StdVideoDecodeH264Mvc;
+            typedef uint64_t StdVideoH265ProfileIdc;
+            typedef uint64_t StdVideoH265Level;
+            typedef uint64_t StdVideoH265VideoParameterSet;
+            typedef uint64_t StdVideoH265SequenceParameterSet;
+            typedef uint64_t StdVideoDecodeH265PictureInfo;
+            typedef uint64_t StdVideoDecodeH265ReferenceInfo;
+            typedef uint64_t StdVideoEncodeH264RefMemMgmtCtrlOperations;
+            typedef uint64_t StdVideoEncodeH264SliceHeader;
+            typedef uint64_t StdVideoH265PictureParameterSet;
+            typedef uint64_t StdVideoEncodeH265PictureInfo;
+            typedef uint64_t StdVideoEncodeH265SliceSegmentHeader;
+            typedef uint64_t StdVideoEncodeH264ReferenceInfo;
+            typedef uint64_t StdVideoEncodeH264PictureInfo;
+            typedef uint64_t StdVideoEncodeH265ReferenceModifications;
+            typedef uint64_t VkAccelerationStructureMotionInstanceDataNV;
+            typedef uint64_t StdVideoEncodeH265ReferenceInfo;
+
+            namespace agi {
+            namespace replay2 {
+
+            class VkStructFactory {
+            public:
+                virtual ~VkStructFactory() {}
+                virtual size_t VkStructMemorySize() = 0;
+                virtual size_t PNextChainMemorySize() = 0;
+            };
+
+        """))
+
+        for type_name in vulkan_metadata.types.basetypes:
+            remapper_h.write("typedef uint64_t " + type_name + ";\n")
+        remapper_h.write("\n")
+
+        for type_name in vulkan_metadata.types.handles:
+            remapper_h.write("typedef uint64_t " + type_name + ";\n")
+        remapper_h.write("\n")
+
+        for type_name in vulkan_metadata.types.handle_aliases:
+            remapper_h.write("typedef uint64_t " + type_name + ";\n")
+        remapper_h.write("\n")
+
+        for type_name in vulkan_metadata.types.funcpointers:
+            remapper_h.write("typedef uint64_t " + type_name + ";\n")
+        remapper_h.write("\n")
+
+        for type_name in vulkan_metadata.types.bitmasks:
+            remapper_h.write("typedef uint64_t " + type_name + ";\n")
+        remapper_h.write("\n")
+
+        for type_name in vulkan_metadata.types.bitmask_aliases:
+            remapper_h.write("typedef uint64_t " + type_name + ";\n")
+        remapper_h.write("\n")
+
+        for type_name in vulkan_metadata.types.enums:
+            remapper_h.write("typedef uint64_t " + type_name + ";\n")
+        remapper_h.write("\n")
+
+        for type_name in vulkan_metadata.types.enum_aliases:
+            remapper_h.write("typedef uint64_t " + type_name + ";\n")
+        remapper_h.write("\n")
+
+        for struct in vulkan_metadata.types.structs:
+            remapper_h.write("class " + struct + " {};\n")
+        remapper_h.write("\n")
+
+        all_vulkan_structs = vulkan_metadata.types.structs
+        for struct in sort_structs_dep_order(vulkan_metadata):
```

Comment:
I don't want to repeat myself but I *really* find it hard to understand what being generated here.

Should we at least for this one try to make it with fstrings?